experimental/token/cursor.go

```diff
@@ -77,8 +77,8 @@ func (c *Cursor) IsSynthetic() bool {
 // Clone returns a copy of this cursor, which allows performing operations on
 // it without mutating the original cursor.
 func (c *Cursor) Clone() *Cursor {
-	c2 := *c
-	return &c2
+	var clone Cursor = *c
```

Comment:
Why switch from `:=` to `var`?
```suggestion
	clone := *c
```